src/components/controller/AppController.js

```diff
@@ -1,49 +1,162 @@
-            });
-        let sector_name = this.state.figData.find((sector) => {
-            return this.props.params.sector == sector.slugSector; 
-        }).sector;
-
-        this.setState({indicatorData:indicator_data, sectorName:sector_name, sectorSelected:this.props.params.sector});
+    componentWillMount(){    
+        this.setCategoryName();
+        this.setRecordData();
+        this.setState({heirarchyLevel : this.props.params.heirarchy_level,
+            categorySlug : this.props.params.category,
+            recordSlug : this.props.params.record,
+            series : this.props.route.panelName});
+        this.setNotes();
+        this.setAdditionalDetails();
     }
 
-    componentDidUpdate(prevProps, prevState) {
-        const indicator_data = this.state.figData.find((sector) => {
-                return this.props.params.sector == sector.slugSector;
-            }).subIndicators.find((indicator) => {
-                return this.props.params.indicator == indicator.slugIndicator;
-            });
-        let sector_name = this.state.figData.find((sector) => {
-            return this.props.params.sector == sector.slugSector; 
-        }).sector;
+    componentDidUpdate(prevProps, prevState) {    
+        //
+        if(prevState.categorySlug != this.props.params.category && this.props.params.heirarchy_level == 2){
+            this.setCategorySlug();
+            this.setCategoryName();
+        }
+
+        if(prevState.recordSlug != this.props.params.record || prevState.categorySlug != this.props.params.category){
+            this.setRecordSlug();
+            this.setRecordData();
+        }
+
+        if(prevState.categorySlug != this.props.params.category || prevState.recordSlug != this.props.params.record || prevState.heirarchyLevel != this.props.params.heirarchy_level){
+            this.setNotes();
+        }
+        
+        if(prevProps.params.category != this.props.params.category && this.state.heirarchyLevel == 2){
+            this.setAdditionalDetails();
+        }
         
-        if(prevState.indicatorData != indicator_data){
-            this.setState({indicatorData:indicator_data, sectorName:sector_name, sectorSelected:this.props.params.sector});
+        if(prevProps.params.record != this.props.params.record && this.state.heirarchyLevel == 1){
+            this.setAdditionalDetails();
+        }
+    }
+    setCategorySlug(){
+        this.setState({categorySlug:this.props.params.category});
+    }
+
+    setRecordSlug(){
+        this.setState({recordSlug:this.props.params.record});
+    }
+
+    setRecordData(){
+        let currentState = this.state;
+        if(this.props.params.heirarchy_level == 2 ){
+            let recordData = appController.dataseries.find(function(series){
+                return series.title_slug == currentState.series;
+                }).data.find(function(category){
+                    return category.category_slug == currentState.categorySlug;       
```

Comment:
Camel Case has been used for React app variables. While the variable style for external variables(external data) used is different than the one used for app.